torch/csrc/utils/python_arg_parser.h

```diff
@@ -658,6 +658,9 @@ inline std::vector<int64_t> PythonArgs::intlistWithDefault(
         } catch (std::exception& e) {
           throw_intlist_exception(this, i, obj, idx, e);
         }
+      } else if (torch::is_symint(py::handle(obj))) {
+        res[idx] = py::cast<c10::SymInt>(py::handle(obj))
+                       .guard_int(__FILE__, __LINE__);
```

Comment:
More for my own understanding: instead of silently specializing whenever we plumb a symint to an op who's schema specifies `int[]`, wouldn't it be better to just change all of our schemas to take in `SymInt[]`? (and until then, raising an error makes it easier to identify and fix the places where this happens?)

One reason the guards here are kinda painful is that (I don't think) we get cpp stacktraces in the guard provenance tracking, so it won't be immediately clear to people who sees guards that the guards are coming from our unfixed schemas.

Reply:
oh unless this change is purely for BC reasons - in theory we could change all aten schemas today, but I guess for custom ops, we used to specialize by default whereas now (without this PR) we now error by default on custom op schemas that take in `int[]`.

Reply:
So what we specifically are guarding against here is when Dynamo thinks an int is dynamic (most commonly because dynamic=True) but it actually is something like dim= and cannot possibly be dynamic.

I think modifying the schemas would not be the right thing to do, because there isn't any situation we will actually support dynamism for real, so it is wasted work. In the early days, we errored instead of specializing to make it easier to catch missing symints in schema, but we now have good logging and good coverage so it is more important to avoid erroring eg as in the linked issues